core/src/main/java/com/linecorp/armeria/server/docs/TypeSignatureType.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+/**
+ * Represents the different type that is used for {@link DocService}.
+ */
+public enum TypeSignatureType {
+
+    /**
+     * Base type.
+     */
+    BASE(false, false),
+
+    /**
+     * Named type.
+     */
+    NAMED(false, true),
+
+    /**
+     * Enum type.
+     */
+    ENUM(false, true),
+
+    /**
+     * Iterable type.
+     */
+    ITERABLE(true, false),
+
+    /**
+     * Map type.
+     */
+    MAP(true, false),
+
+    /**
+     * Optional type.
+     */
+    OPTIONAL(true, false),
+
+    /**
+     * Container type.
+     */
+    CONTAINER(true, false),
```

Comment:
What is an example for a container type? Maybe generics? Like ```Id<T>``` ?

Reply:
One example would be:
https://github.com/line/armeria/blob/6bd02dc74ba924ab51740f81adffefccd0351d8e/core/src/test/java/com/linecorp/armeria/internal/server/annotation/AnnotatedDocServiceTest.java#L518

Of course, we cannot represent this type with JSON schema so I'm thinking we just need to omit and add `"additionalProperties": "true"`